Use of `XrSwapchainCreateFlags` for `OpenXRCompositionLayer`

bnjmntmm opened the pull request:
Introduces SwapchainCreateFlags enum to control swapchain creation (NORMAL, STATIC, PROTECTED) in OpenXR composition layers.
This allows the user to set  SwapChainCreate Flags to either have one static image on the composition layer or set the protected content flag which allows stuff like Widevine DRM protected videos to be displayed on the composition layer


As this is the first time **ever** creating something with c++, i am open for feedback, will try my best to answer to your guys' feedback and improve the code. 

A question I have already: Should I rename the values in the enum, as "normal", "static" and "protected" are not that informative and "normal" is just not normal. it just sets a 0 as bit

Thanks ^^

bnjmntmm committed:
Add swapchain create flags support to OpenXR layers

- Introduced SwapchainCreateFlags enum to control swapchain creation (NORMAL, STATIC, PROTECTED) in OpenXR composition layers.
- Allows creation of static layers (never change after creation) and protected layers (for DRM-protected content).
- Changed internal logic from enum to bool for simplicity; users now select "protected" or not.
- Added support for protected content in viewport-based layers.
- Refactored naming, documentation, and improved reusability.
- Minor cleanup: removed unused variable and added missing space.

dsnopek commented:
Thanks!

I don't think it makes sense to have "static" as an option, because we already support that for `Viewport`-based composition layers automatically, and I don't think there is really a use case for it with Android surfaces. Just adding the ability to toggle "protected" should be enough.

Also, you need to make sure that this flag is used for `Viewport`-based layers as well. That would be done in `OpenXRViewportCompositionLayerProvider::update_and_acquire_swapchain()`

bnjmntmm commented:
I've changed the enum to a simple bool. Also tested it and it still works. Off = pink screen, On = video is shown
Also added some basic documentation but that can be changed however you guys like :D

Also a question regarding the viewport based layers. Is there even the need to create a protected layer? I don't think it can be used properly but idk. I just added an extra if.
Maybe the if should be an extra check so it can't be static and protected.. idk if that even works

bnjmntmm commented:
I think I have to change something in this pr.. as I rebased based on the docs and now it wanna also merge the other files.

dsnopek reviewed (commented):
Thanks!

At a high-level, this is looking good. I've mostly got notes about naming, and one functional note with regard to recreating the swapchain when the source of the content is a subviewport

Oh, and in the end, you'll need to squash everything into a single commit per Godot's pull request worflow. See https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase

bnjmntmm commented on modules/openxr/extensions/openxr_composition_layer_extension.cpp:
```diff
@@ -489,6 +495,9 @@ bool OpenXRViewportCompositionLayerProvider::update_and_acquire_swapchain(bool p
 	if (p_static_image) {
 		create_flags |= XR_SWAPCHAIN_CREATE_STATIC_IMAGE_BIT;
 	}
+	if (swapchain_state.use_protected_content) {
```
Would that just work if I do a check like `if (swapchain_size == subviewport.viewport_size && !p_static_image && !subviewport.static_image && !subviewport.protected_content) {` ?

m4gr3d reviewed (approved):
Haven't tested the changes but the code looks good.

bnjmntmm commented:
What does the one failed Linux / Editor with Mono mean? 
Will squash later in the evening

dsnopek reviewed (commented):
Thanks, this looks great! Just two tiny notes, and this needs to be squashed, but after that I'd be happy to approve :-)

dsnopek commented on modules/openxr/scene/openxr_composition_layer.cpp:
```diff
@@ -113,6 +113,9 @@ void OpenXRCompositionLayer::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("is_natively_supported"), &OpenXRCompositionLayer::is_natively_supported);
 
+	ClassDB::bind_method(D_METHOD("is_protected_content"), &OpenXRCompositionLayer::is_protected_content);
+	ClassDB::bind_method(D_METHOD("set_protected_content", "use_protected_content"), &OpenXRCompositionLayer::set_protected_content);
```
```suggestion
	ClassDB::bind_method(D_METHOD("set_protected_content", "protected_content"), &OpenXRCompositionLayer::set_protected_content);
```

bnjmntmm commented:
I'm stupid.. forgot to remove the old part in the docs

bnjmntmm commented:
OKay.. squashing annoyed me so much because it reverted some changes, but I think every proposed change is in there.. I hope i didn't miss anything.

dsnopek reviewed (approved):
Thanks! I haven't tested, but the code looks great to me :-)

bnjmntmm commented on modules/openxr/scene/openxr_composition_layer.cpp:
```diff
@@ -410,6 +414,13 @@ bool OpenXRCompositionLayer::is_natively_supported() const {
+void OpenXRCompositionLayer::set_protected_content(bool p_protected_content) {
+	openxr_layer_provider->set_protected_content(p_protected_content);
+}
+bool OpenXRCompositionLayer::is_protected_content() const {
```
need to squash commits now
